### PR TITLE
Removed username and improved validation

### PR DESCRIPTION
**Pull request by freaksN**

### CODE REVIEW

**Comment by dennis173** [comment]
In the registration from, it would be awesome if the "password must match" warning disappears as soon as the user types again in the fields.

File: src/features/registration/components/RegistrationForm.tsx [hunk 1]
```diff
@@ -59,12 +85,28 @@ const RegistrationForm = (): JSX.Element => {
...
       />
+
+      <TextField
+        helperText={'Please agree to terms of service and privacy policy'}
+        error={!termsAndPolicies.checkedA}
+        style={{ marginTop: '-33px' }}
+        InputProps={{
+          disableUnderline: true,
+        }}
+      />
```
Comment (niweb) [inline src/features/registration/components/RegistrationForm.tsx]: This seems to be quite a hack: Rendering a whole TextField just for the purpose of showing an error message when the Checkbox above is not checked.

File: src/features/registration/components/RegistrationForm.tsx [hunk 2]
```diff
@@ -59,12 +85,28 @@ const RegistrationForm = (): JSX.Element => {
         type="password"
         name="passwordRepeat"
         helperText={retypePasswordHelperText}
-        error={areRegistrationFieldsFalse}
+        error={retypePassword}
       />
       <FormControlLabel
```
Comment (niweb) [inline src/features/registration/components/RegistrationForm.tsx]: In the last PRs I authored I also included a Checkbox-Component, that could probably also be of use here.

File: src/features/registration/components/RegistrationForm.tsx [hunk 3]
```diff
@@ -15,35 +16,60 @@ interface RegistrationFormValues extends InputValues {
...
+  const handleChange = name => event => {
+    setTermsAndPolicies({ ...termsAndPolicies, [name]: event.target.checked })
+  }
 
   if (success) {
-    return <Notification type={NotificationType.success} message="Success!" />
+    return (
+      <div>
+        <Notification type={NotificationType.success} message="Success!" />
+        <LoginForm />
```
Comment (niweb) [inline src/features/registration/components/RegistrationForm.tsx]: After Registering, logging in, then out again, I am not able to reach the registration form anymore until I reload the website. This is probably an issue with storing the success-state in the redux store. Should be fixed later on.

**Pull request merged**